### DISCUSSION TITLE
Liquid Styleguide

### DISCUSSION THREAD

**Discussion by tlegasse** [Shopify/prettier-plugin-liquid#161] [Q&A]
I have a use-case for dynamic tag types. E.g. wrapping elements in conditional tags such that either the `predictive-search` web-component, or a standard `form` tag is used. There's no detectible downside to doing so to me, however, the Prettier Liquid plugin throws Liquid HTML parsing errors. I'd love to know if there's a general Liquid styleguide that Shopify uses internally so I can better understand:

1. What should be expected to cause Liquid HTML parsing errors
2. What is considered best practice

Example code that causes errors: 
```
      {%- if section.settings.show_predictive_input -%}
        <predictive-search data-loading-text="{{ 'accessibility.loading' | t }}">
      {%- endif -%}
```
Thanks!

**Comment by charlespwd** [comment]
:wave: This is because of https://github.com/Shopify/prettier-plugin-liquid/issues/90 

Fix is hard to get right because of HTML whitespace handling and striking the balance between flagging "you forgot to close your tag" with "intentional unclosed tag inside if statements."

**Reply by tlegasse** [reply]
Thanks so much for the context! Very happy to be using the plugin.
